From: SeongJae Park <sj@kernel.org>
Date: Mon, 7 Oct 2024 09:42:17 -0700
Subject: [PATCH] ==== remove DAMON debugfs interface ====

[RFC PATCH] mm/damon: remove DAMON debugfs interface

DAMON debugfs interface was the only user interface of DAMON at the
beginning[1].  However, it turned out the interface would be not good
enough for long-term flexibility and stability.

In Feb 2022[2], we therefore introduced DAMON sysfs interface as an
alternative user interface that aims long-term flexibility and
stability.  With its introduction, DAMON debugfs interface has announced
to be deprecated in near future.

In Feb 2023[3], we announced the official deprecation of DAMON debugfs
interface.  In Jan 2024[4], we further made the deprecation difficult to
be ignored.

And as of this writing (2024-10-14), no problem or concerns about the
deprecation have reported.  Apparently users are already moved to the
alternative, or made good plans for the change.

Remove the DAMON debugfs interface code from the tree.  Given the past
timeline and the absence of reported problems or concerns, it is safe
enough to be done.  That said, we will not drop the RFC tag of this
patch series at least until the end of this year, to use this as the
real last call for users.

[1] https://lore.kernel.org/20210716081449.22187-1-sj38.park@gmail.com
[2] https://lore.kernel.org/20220228081314.5770-1-sj@kernel.org
[3] https://lore.kernel.org/20230209192009.7885-1-sj@kernel.org
[4] https://lore.kernel.org/20240130013549.89538-1-sj@kernel.org

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_meta_changes/MmeuEpnX | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 damon_meta_changes/MmeuEpnX

diff --git a/damon_meta_changes/MmeuEpnX b/damon_meta_changes/MmeuEpnX
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

